src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/Server/SqlSer.cs

```diff
@@ -50,16 +52,21 @@ internal static void Serialize(Stream s, object instance)
         //
         // Use a per-thread cache, so that there are no synchronization
         // issues when accessing cache entries from multiple threads.
-        [ThreadStatic]
-        private static Hashtable s_types2Serializers;
+        private static ConcurrentDictionary<Type, Serializer> s_types2Serializers;
 
         private static Serializer GetSerializer(Type t)
         {
+            Serializer s = null;
             if (s_types2Serializers == null)
-                s_types2Serializers = new Hashtable();
+            {
+                s_types2Serializers = new ConcurrentDictionary<Type, Serializer>();
+            }
 
-            Serializer s = (Serializer)s_types2Serializers[t];
-            if (s == null)
+            if (s_types2Serializers.ContainsKey(t))
```

Comment:
This `if-else` block could be replaced with the following block:
```
if(!s_types2Serializers.TryGetValue(t, out s))
{
    s = GetNewSerializer(t);
    s_types2Serializers[t] = s;
}
```

Reply:
TryGetValue will return Default value if there is no value for the key, condition will never be false and compiler will not enter if block. Although, I think I should keep the check if(s==null) after contains key check because we don't check  null value before entry to dictionary.